spreadsheets.js

```diff
@@ -422,8 +438,103 @@ function update_schedule(service_account_auth, key, colname, format, schedule, c
     });
 }
 
+// parse the input string for a results update
+function parse_result(input_string){
+    var tokens = get_tokens(input_string);   
+    var result = find_result(tokens);
+    var players = find_players(tokens);
+    
+    return {
+        "white": players.white,
+        "black": players.black,
+        "result": result,
+    };
+}
+
+function get_tokens(input_string){
+    return input_string.split(" ");
+}
+
+function find_result(tokens){
+    var result;
+    _.some(tokens, function(token){
+        return (result = VALID_RESULTS[token.toUpperCase()]) ? true : false;
+    });
+    return result;
+}
+
+function find_players(tokens){
+    // in reality, the order is arbitrary.
+    // just assuming first I find is white, the second is black
+    var players = {};
+    
+    var player_tokens = filter_player_tokens(tokens);
+
+    //assuming we found 2 tokens, we should convert them to player names
+    if(player_tokens.length == 2){
+        players.white = player_tokens[0];
+        players.black = player_tokens[1];
+    }
+
+    return players;
+}
+
+function filter_player_tokens(tokens){
```

Comment:
This, I like this.   I had been considering doing something similar, but never got around to looking for the documentation on which characters a slack username ID might contain. I just tried looking it up again to see if I could verify this regex, but couldn't find it. 

Did you find documentation on which characters the usernames can contain?